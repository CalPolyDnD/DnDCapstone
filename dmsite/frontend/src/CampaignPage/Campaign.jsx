import React from 'react';
import {
  ListGroup, ListGroupItem, Card, CardBody, Button, CardHeader,
} from 'reactstrap';
import './style.css';
import axios from 'axios';

const FETCH_CURRENT_USER_URL = 'http://localhost:8000/rest-auth/user/';
const FETCH_CAMPAIGNS_URL = 'http://localhost:8000/get_campaigns';

class Campaign extends React.Component {
  constructor() {
    super();
    this.state = {
      campaigns: [],
    };
  }

  onClick() {
    this.props.history.push(`/a/${this.props.a}`);
    window.location.reload();


  }

  componentDidMount() {
    axios.get(FETCH_CURRENT_USER_URL)
      .then((userRes) => {
        axios.post(FETCH_CAMPAIGNS_URL, {
          user: userRes.data.email,
        })
          .then((campRes) => {
            this.setState({ campaigns: campRes.data });
          });
      });
  }

  getCampaignList() {
    const jsonObj = [];
    const { campaigns } = this.state;
    const { history } = this.props;
    for (let count = 0; count < campaigns.length; count += 1) {
      const result = campaigns[count];
      jsonObj.push(
        <ListGroupItem
          onClick={() => { history.push(`/home${campaigns.name}`); }}
          style={{ backgroundColor: '#3d3d3d', color: 'white' }}
        >
          {result.name}
        </ListGroupItem>,
      );
    }
    return jsonObj;
  }

  render() {
    return (
      <div id="div" style={{ paddingTop: '5%' }}>
        <Card id="card" style={{ borderWidth: 0 }}>
          <CardHeader tag="h3" align="center" style={{ backgroundColor: '#303030', color: 'white' }}>Campaigns</CardHeader>
          <CardBody style={{ backgroundColor: '#3d3d3d', borderWidth: 0 }}>
            <ListGroup align="center" className="campaign-list" flush>
              {this.getCampaignList()}
            </ListGroup>
            <Button color="primary" size="md" className="btn-block mt-3">Create New Campaign</Button>
            <Button color="secondary" size="md" className="btn-block mt-3"
                    disabled={this.state.campaigns.length === 0}
                    onClick={this.onClick}>Save & Continue </Button>
          </CardBody>
        </Card>
      </div>
    );
  }
}

export default Campaign;
