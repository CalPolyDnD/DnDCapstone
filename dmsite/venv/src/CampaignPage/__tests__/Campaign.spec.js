import React from 'react';
import renderer from 'react-test-renderer';
import {
  ListGroup, ListGroupItem, Card, CardBody, Button, CardHeader,
} from 'reactstrap';
import Campaign from '../Campaign';

it('renders correctly', () => {
  const tree = renderer
    .create(<Campaign />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
