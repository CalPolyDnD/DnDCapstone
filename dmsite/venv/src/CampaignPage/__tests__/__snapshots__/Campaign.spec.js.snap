// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="div"
  style={
    Object {
      "alignSelf": "center",
      "display": "flex",
      "justifyContent": "center",
      "paddingTop": "5%",
      "width": "100%",
    }
  }
>
  <div
    className="card"
    id="card"
    style={
      Object {
        "alignSelf": "center",
        "borderWidth": 0,
        "display": "flex",
        "justifyContent": "center",
        "width": "50%",
      }
    }
  >
    <h3
      align="center"
      className="card-header"
      style={
        Object {
          "backgroundColor": "#303030",
          "color": "white",
        }
      }
    >
      Campaigns
    </h3>
    <div
      className="card-body"
      style={
        Object {
          "backgroundColor": "#3d3d3d",
          "borderWidth": 0,
        }
      }
    >
      <ul
        align="center"
        className="campaign-list list-group list-group-flush"
      >
        <div
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#303030",
              "display": "flex",
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          <li
            className="list-group-item"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#3d3d3d",
                "color": "white",
              }
            }
          >
            Campaign 1
          </li>
          <li
            className="list-group-item"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#3d3d3d",
                "color": "white",
              }
            }
          >
            Campaign 2
          </li>
          <li
            className="list-group-item"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#3d3d3d",
                "color": "white",
              }
            }
          >
            Campaign 3
          </li>
        </div>
      </ul>
      <button
        aria-label={null}
        className="btn-block mt-3 btn btn-primary btn-md"
        onClick={[Function]}
      >
        Create New Campaign
      </button>
    </div>
  </div>
</div>
`;
